packages/testcontainers/src/containers/Hydra/NativeChainContainer.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  GenericContainer,
+  Network,
+  StartedTestContainer
+} from 'testcontainers'
+import fetch from 'cross-fetch'
+import { AbstractStartedContainer } from 'testcontainers/dist/modules/abstract-started-container'
+import { getNetwork, Network as JellyfishNetwork, NetworkName } from '@defichain/jellyfish-network'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+
+/**
+ * Mandatory options to start NativeChain with
+ */
+export interface StartOptions {
+  // TODO(fuxingloh): change to cookie based auth soon
+  user: string
+  password: string
+  timeout?: number
+  startFlags?: StartFlags[]
+}
+
+export interface StartFlags {
+  name: string
+  value: number
+}
+
+/**
+ * DeFiChain NativeChain node managed in docker
+ */
+export class NativeChainContainer extends GenericContainer {
+  /**
+   * @param {string} image docker image name
+   */
+  constructor (
+    image: string = NativeChainContainer.image
+  ) {
+    super(image)
+  }
+
+  static get image (): string {
+    if (process?.env?.DEFICHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.DEFICHAIN_DOCKER_IMAGE
+    }
+    return 'defi/defichain:HEAD-02ef6a1b3'
+  }
+
+  public static readonly DefaultStartOptions = {
+    user: 'testcontainers-user',
+    password: 'testcontainers-password'
+  }
+
+  protected startOptions: StartOptions = NativeChainContainer.DefaultStartOptions
+
+  public withStartOptions (startOptions: StartOptions): this {
+    this.startOptions = startOptions
+    return this
+  }
+
+  protected networkConfig: JellyfishNetwork = getNetwork('testnet')
+
+  public withJellyfishNetwork (networkName: NetworkName): this {
```

Comment:
How about `withBlockchianNetwork()` instead of "Jellyfish Network" since jellyfish network is an actual allowed jellyfish chain params on the C++ side.